Content.Server/Imperial/Radio/JobPlayerAndColor.cs

```diff
@@ -0,0 +1,173 @@
+        var prototypes = IoCManager.Resolve<IPrototypeManager>().EnumeratePrototypes<IdColorAndJobPrototype>();
+
+        foreach (var proto in prototypes)
+        {
+            // Добавляем в словарь только уникальные ID
+            if (!SaveColorForJob.ContainsKey(proto.ID))
+            {
+                SaveColorForJob.Add(proto.ID, proto.Color);
+            }
+        }
+    }
+
+    // Словарь для перевода должностей и для должностей состоящих не только из букв
+    private static readonly Dictionary<string, string> JobTranslations = new()
+    {
+        { "Central Commander", "Центральное Командование" },
+        { "Centcom Quarantine Officer", "РХБЗЗ" }
+    };
+
+    private string TranslateJob(string job)
+    {
+        // Ищем перевод в словаре
+        if (JobTranslations.TryGetValue(job, out var translatedJob))
+        {
+            return translatedJob;
+        }
+
+        // Возвращаем оригинал, если перевода нет
+        return job;
+    }
+
+	/// <summary>
+    /// Для EnityUid ищется карта в руках или в пда. Если карта нашлась, мы смотрим какая там работа и достаём её
+    /// </summary>
+    /// <param name="uid"> uid игрока, у которого мы ищем карту</param>
+    /// <returns></returns>
+    public string? GetJobPlayer(EntityUid uid)
+    {
+        // Проверяем нашлась ли карта и какое id у карты
+        if (_idCardSystem.TryFindIdCard(uid, out var id))
+        {
+            // Мы сохраняем работу
+            string? playerJob = id.Comp.LocalizedJobTitle;
+            // Если работа нашлась верно, мы начинаем основной процесс
+            if (playerJob != null)
+            {
+                // Делаем начало должности с заглавной буквы и сохраняем в playerJob
+                playerJob = char.ToUpper(playerJob[0]) + playerJob.Substring(1);
+                // Заменяем тире, дефисы и минусы на пробелы. Эти символы при форматировании ломают вывод сообщения в рацию
+                playerJob = Regex.Replace(playerJob, @"[-–—−]", " ");
+                // Уберём лишние символы "!?", которые могут ломать в целом вывод сообщения в радио канале
+                playerJob = Regex.Replace(playerJob, @"[^a-zA-Zа-яА-ЯёЁ ]", "");
+                // Переводим должность через словарь. Если перевода нет, playerJob не меняется
+                playerJob = TranslateJob(playerJob);
+                return playerJob.Trim();
+
+            }
+
+
+        }
+        // Если работы нет, то возвращается должность "Неизвестно"
+        return "Неизвестно";
+    }
+
+    /// <summary>
+    /// Метод, который отвечает за подбор цвета для должности. Используется словарь, который работает по О(1), что быстрее if и подобного
+    /// </summary>
+    /// <param name="jobPlayer"> Нужно получить из метода GetJobPlayer. Или же вставляйте сюда свою string? работу, но нужно привести её к нижниму регистру для словаря. Метод ваш_стринг.ToLower()</param>
+    /// <returns></returns>
+    public string? GetColorPlayer(string? jobPlayer)
```

Comment:
Не вижу ни одной причины возвращать нуллейбл. По идее этот метод должен уже точно определять цвет профессии, что бы не было лишних null проверок
```suggestion
    public string GetColorPlayer(string? jobPlayer)
```

Reply:
Мышка в рацию сказала сквик, у неё функция по поиску jobTitle через pda выдаст null, так как у мыши её нет

Не использовать тут string? - будет создавать фатальные ошибки...

### Креатив, очень прошу быть вдумчивым в ревью, чтобы не предлагать людям то, что сломает наоборот код

Закрываю с меткой ### "НИ В КОЕМ СЛУЧАЕ ТАК НЕ ДЕЛАТЬ!"

Reply:
Мышка в рацию сказала сквик, и эта функция должна вернуть цвет игрока (исходя из имени), а не *может быть* цвет игрока. Если мышка сказала сквик, то пусть уж эта функция возвращает как раз таки дефолтный цвет (что я так же отмечал в ревью)